src/main/java/spoon/support/StandardEnvironment.java

```diff
@@ -274,7 +274,7 @@ public void report(Processor<?> processor, Level level, String message) {
 	}
 
 	private void print(String message, Level messageLevel) {
-		if (messageLevel.toInt() <= this.level.toInt()) {
+		if (this.level.compareTo(Level.OFF) > 0 && messageLevel.toInt() <= this.level.toInt()) {
```

Comment:
Perhaps this is more clear?

```suggestion
		if (this.level != Level.OFF && messageLevel.toInt() <= this.level.toInt()) {
```

I think it's a bit easier to read as it doesn't require one to go check the declaration order of the `Level` values. This is probably a matter of preference, though, if you prefer your solution feel free to keep it.

Reply:
Note that there's a tiny behavioral difference: `level != Level.OFF` will cause the if-block not to be entered even if the passed level is `OFF`, whereas your original solution will cause it to be entered (but do nothing as no case in the switch matches)

Reply:
sure, that's better, I didn't know enums could compare like that (that's how little java I know 😅)